core/src/builtin_widgets.rs

```diff
@@ -999,3 +1001,69 @@ impl<T> std::ops::DerefMut for FatObj<T> {
   #[inline]
   fn deref_mut(&mut self) -> &mut Self::Target { &mut self.host }
 }
+
+/// DeclarerWithSubscription, a declarer with subscriptions
+///
+/// Used to wraps a declarer to make it the widget auto unsubscribe when
+/// disposed. Normally you should not use this directly, most Widget types
+/// derive with Declare attribute has support builtin widgets has the ability
+/// of unsubscribing when disposed.
+pub struct DeclarerWithSubscription<T> {
+  inner: T,
+  subscribes: Vec<BoxSubscription<'static>>,
```

Comment:
Perhaps utilizing `SmallVec` would be a better option.


```suggestion
  subscribes: SmallVec<[BoxSubscription<'static>;1]>,
```